controller/src/main/java/io/pravega/controller/task/KeyValueTable/TableMetadataTasks.java

```diff
@@ -0,0 +1,267 @@
+        this.requestTracker = requestTracker;
+    }
+
+    @Synchronized
+    public void initializeStreamWriters(final EventStreamClientFactory clientFactory,
+                                        final String streamName) {
+        this.requestStreamName = streamName;
+        requestEventWriterRef.set(clientFactory.createEventWriter(requestStreamName,
+                ControllerEventProcessors.CONTROLLER_EVENT_SERIALIZER,
+                EventWriterConfig.builder().build()));
+        writerInitFuture.complete(null);
+    }
+
+    /**
+     *  Create a Key-Value Table.
+     *
+     * @param scope      scope name.
+     * @param kvtName    KVTable name.
+     * @param kvtConfig  KVTable configuration.
+     * @param createTimestamp  KVTable creation timestamp.
+     * @return update status.
+     */
+    public CompletableFuture<CreateKeyValueTableStatus.Status> createKeyValueTable(String scope, String kvtName,
+                                                                                   KeyValueTableConfiguration kvtConfig,
+                                                                                   final long createTimestamp) {
+        final long requestId = requestTracker.getRequestIdFor("createKVTable", scope, kvtName);
+        return Futures.exceptionallyExpecting(kvtMetadataStore.getState(scope, kvtName, true, null, executor),
+                e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, KVTableState.UNKNOWN)
+                .thenCompose(state -> {
+                    if (state.equals(State.UNKNOWN) || state.equals(State.CREATING)) {
+                        // 1. post event for CreateKVTable.
+                        return  Futures.withCompletion(kvtMetadataStore.checkScopeExists(scope)
+                                .thenCompose(exists -> {
+                                    if (exists) {
+                                        return Futures.exceptionallyExpecting(kvtMetadataStore.createEntryForKVTable(scope, kvtName, executor)
+                                                        .thenCompose(uuid -> {
+                                                            CreateTableEvent event = new CreateTableEvent(scope, kvtName, kvtConfig.getPartitionCount(), createTimestamp, requestId, uuid);
+                                                            return addIndexAndSubmitTask(event,
+                                                                    () -> CompletableFuture.completedFuture(null))
+                                                                    .thenCompose(x -> checkDone(() -> isCreated(scope, kvtName))
+                                                                            .thenApply(y -> CreateKeyValueTableStatus.Status.SUCCESS));
+                                                        }),
+                                                e -> Exceptions.unwrap(e) instanceof StoreException.DataExistsException,
```

Comment:
you cannot get DataExistsException from isCreated method. 
The task will complete. You are checking when the task completes using isCreated method.. Once complete, you should check if the table config and table creation time are as expected by the current request.. if they do, its a tableExists exception.

Reply:
I can get a `DataExistsException` when `createEntryForKVTable()` is invoked and the entry is already found inside the scope. This check guards against that.

Reply:
what are you doing if you get DataExists exception?
secondly, the two phases are separate -- 1. submitting the request, 2. tracking the request.. 
so do the handling of errors from two phases separately so that you know and are able to define handling specific to the phase.

Reply:
`DataExistsException` here indicates that an entry for a KVTable with this name exists in the scopesTables. So we're returning TABLE_EXISTS exception in this case and event for table creation is not sent out.

Reply:
consider the case where two concurrent createTables come.. one gets data exists exception.. 
it still needs to check if the table creation is complete or not.. so it should check the state.. and if state is non-active, then it should wait until the table is created before letting the  know..

Reply:
Done as discussed. pls check....